packages/rest/src/controllers/connections/ConnectionController.ts

```diff
@@ -0,0 +1,188 @@
+import type { ConnectionRecord } from "@aries-framework/core";
+
+import {
+  Agent,
+  ConnectionRepository,
+  ConnectionInvitationMessage,
+  RecordNotFoundError,
+  AgentConfig,
+} from "@aries-framework/core";
+import {
+  NotFoundError,
+  InternalServerError,
+  JsonController,
+  Get,
+  Post,
+  Param,
+  Delete,
+  Body,
+} from "routing-controllers";
+import { Service, Inject } from "typedi";
+
+import { InvitationRequest } from "../../schemas/InvitationRequest";
+
+@JsonController("/connections")
+@Service()
+export class ConnectionController {
+  @Inject()
+  private agent: Agent;
+
+  public constructor(agent: Agent) {
+    this.agent = agent;
+  }
+
+  /**
+   * Retrieve connection record by connectionId
+   */
+  @Get("/:connectionId")
+  public async getConnectionById(@Param("connectionId") connectionId: string): Promise<ConnectionModel> {
+    const connection = await this.agent.connections.findById(connectionId);
+
+    if (!connection) {
+      throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`);
+    }
+
+    return this.mapConnection(connection);
+  }
+
+  /**
+   * Retrieve all connections records
+   */
+  @Get("/")
+  public async getAllConnections(): Promise<ConnectionModel[]> {
+    const connections = await this.agent.connections.getAll();
+
+    return connections.map((conn) => this.mapConnection(conn));
+  }
+
+  /**
+   * Creates a new ConnectionRecord and InvitationMessage.
+   * Returns ConnectionRecord with invitation and invitation_url
+   */
+  @Post("/create-invitation")
+  public async createInvitation(
+    @Body()
+    invitationRequest?: InvitationRequest
+  ): Promise<{
+    invitation_url: string;
+    invitation: ConnectionInvitationMessage;
+    connection: ConnectionModel;
+  }> {
+    try {
+      const { invitation, connectionRecord } = await this.agent.connections.createConnection(invitationRequest);
+
+      const config = this.agent.injectionContainer.resolve(AgentConfig);
+
+      const connection = this.mapConnection(connectionRecord);
+
+      return {
+        invitation_url: invitation.toUrl({ domain: config.endpoints[0] + "/invitation", useLegacyDidSovPrefix: true }),
+        invitation: invitation,
+        connection: connection,
+      };
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`mediator with mediatorId ${invitationRequest?.mediatorId} not found`);
+      }
+      throw new InternalServerError(`something went completely wrong: ${error}`);
+    }
+  }
+
+  /**
+   * Receive connection invitation as invitee and create connection. If auto accepting is enabled
+   * via either the config passed in the function or the global agent config, a connection
+   * request message will be send.
+   */
+  @Post("/receive-invitation")
+  public async receiveInvitation(@Body() invitation: ConnectionInvitationMessage): Promise<ConnectionModel> {
```

Comment:
There is no options to specify whether you want to use auto acceptance for connections. I'm fine with not adding this intially but would be good to open a issue to keep track of these things